docs(pdb): updates to handling of pod disruption budget with template config (#8427)
strimzi/strimzi-kafka-operator: documentation/modules/managing/proc-drain-cleaner-deploying.adoc

[role="_abstract"] 
Deploy the Strimzi Drain Cleaner to the Kubernetes cluster where the Cluster Operator and Kafka cluster are running.

Strimzi sets a default `PodDisruptionBudget` (PDB) that allows only one Kafka or ZooKeeper pod to be unavailable at any given time.
To use the Drain Cleaner for planned maintenance or upgrades, you must set a PDB of zero.
This is to prevent voluntary evictions of pods, and ensure that the Kafka or ZooKeeper cluster remains available.
You do this by setting the `maxUnavailable` value to zero in the `Kafka` or `ZooKeeper` template. 
`StrimziPodSet` custom resources manage Kafka and ZooKeeper pods using a custom controller that cannot use the `maxUnavailable` value directly.
Instead, the `maxUnavailable` value is converted to a `minAvailable` value.

> If the `maxUnavailable` property is set to `0` (zero), the `minAvailable` setting is also `0` (zero), allowing zero pods to be unavailable.

Reviewer: ```suggestion
If there are three broker pods and the `maxUnavailable` property is set to `0` (zero), the `minAvailable` setting is `3`, requiring all three broker pods to be available and allowing zero pods to be unavailable.
```

.Procedure

> . Configure a pod disruption budget (PDB) of `0` (zero) for your Kafka deployment using `template` settings in the `Kafka` resource.

Reviewer: ```suggestion
. Set `maxUnavailable` to `0` (zero) in the Kafka and ZooKeeper sections using `template` settings in the `Kafka` resource.
```

  # ...
----
+

> A PDB of zero prevents the automatic eviction of pods in case of planned disruptions,

Reviewer: ```suggestion
This setting prevents the automatic eviction of pods in case of planned disruptions,
```
